From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Euphyllia Bierque <bierque.euphyllia@gmail.com>
Date: Sun, 15 Dec 2024 22:32:09 +0100
Subject: [PATCH] Add TPS From Region


diff --git a/src/main/java/org/bukkit/Bukkit.java b/src/main/java/org/bukkit/Bukkit.java
index 8ab94f8189ebd9d4158231871abdebec399deb2c..5571430fd1b9cae2b4e5d634f6d5560792aeeeb8 100644
--- a/src/main/java/org/bukkit/Bukkit.java
+++ b/src/main/java/org/bukkit/Bukkit.java
@@ -2429,6 +2429,30 @@ public final class Bukkit {
     }
     // Paper end
 
+    // Folia start
+    /**
+     * Gets the current location TPS.
+     *
+     * @param location the location for which to get the TPS
+     * @return current location TPS (5s, 15s, 1m, 5m, 15m in Folia-Server), or null if the region doesn't exist
+     */
+    @Nullable
+    public static double[] getTPS(@NotNull Location location) {
+        return server.getTPS(location);
+    }
+
+    /**
+     * Gets the current chunk TPS.
+     *
+     * @param chunk the chunk for which to get the TPS
+     * @return current chunk TPS (5s, 15s, 1m, 5m, 15m in Folia-Server), or null if the region doesn't exist
+     */
+    @Nullable
+    public static double[] getTPS(@NotNull Chunk chunk){
+        return server.getTPS(chunk);
+    }
+    // Folia end
+
     /**
      * Get the advancement specified by this key.
      *
diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index ad816538b30079c62d5e1eb98c6f4b61e12e8d47..d52f642aadea9ec19a769f43d995f9c6fa0934fb 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -2076,6 +2076,26 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
     double getAverageTickTime();
     // Paper end
 
+    // Folia start
+    /**
+     * Gets the current location TPS.
+     *
+     * @param location the location for which to get the TPS
+     * @return current location TPS (5s, 15s, 1m, 5m, 15m in Folia-Server), or null if the region doesn't exist
+     */
+    @Nullable
+    public double[] getTPS(@NotNull Location location);
+
+    /**
+     * Gets the current chunk TPS.
+     *
+     * @param chunk the chunk for which to get the TPS
+     * @return current chunk TPS (5s, 15s, 1m, 5m, 15m in Folia-Server), or null if the region doesn't exist
+     */
+    @Nullable
+    public double[] getTPS(@NotNull Chunk chunk);
+    // Folia end
+
     // Paper start
     /**
      * Gets the active {@link org.bukkit.command.CommandMap}
